Fst2List.cpp

```diff
@@ -2361,19 +2822,42 @@ int main_Fst2List(int argc, char* const argv[]) {
         case 's': aa.prMode = PR_SEPARATION; break;
         case 'a': aa.prMode = PR_TOGETHER; break;
         default:
-          error("Invalid arguments: rerun with --help\n");
+          error("Invalid output format (-f), valid values are \"a\" or \"s\": rerun with --help\n");
           return USAGE_ERROR_CODE;
       }
       break;
     case 'd':
       aa.enableLoopCheck = false;
       break;
+    case 'D' :
+    	// load morphological dictionaries
+      if (options.vars()->optarg[0]!='\0') {
+        if (strcmp(morpho_dic, "") == 0) {
```

Comment:
Refactor with:
**if (strcmp(morpho_dic, "") != 0)**
so that:
**strcpy(morpho_dic, options.vars()->optarg);**
is not duplicated